org_fedora_oscap/scap_content_handler.py

```diff
@@ -0,0 +1,205 @@
+#
+# Copyright (C) 2021 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+# Red Hat Author(s): Jan Černý <jcerny@redhat.com>
+#
+
+from collections import namedtuple
+import os
+import xml.etree.ElementTree as ET
+
+from org_fedora_oscap.content_handling import parse_HTML_from_content
+
+# namedtuple class (not a constant, pylint!) for info about a XCCDF profile
+# pylint: disable-msg=C0103
+ProfileInfo = namedtuple("ProfileInfo", ["id", "title", "description"])
+
+ns = {
+    "ds": "http://scap.nist.gov/schema/scap/source/1.2",
+    "xccdf-1.1": "http://checklists.nist.gov/xccdf/1.1",
+    "xccdf-1.2": "http://checklists.nist.gov/xccdf/1.2",
+    "xlink": "http://www.w3.org/1999/xlink"
+}
+
+
+class SCAPContentHandlerError(Exception):
+    """Exception class for errors related to SCAP content handling."""
+    pass
+
+
+class SCAPContentHandler:
+    def __init__(self, file_path, tailoring_file_path=None):
+        """
+        Constructor for the SCAPContentHandler class.
+
+        :param file_path: path to an SCAP file (only SCAP source data streams,
+        XCCDF files and tailoring files are supported)
+        :type file_path: str
+        :param tailoring_file_path: path to the tailoring file, can be None if no tailoring exists
+        :type tailoring_file_path: str
+        """
+        self.file_path = file_path
+        tree = ET.parse(file_path)
+        self.root = tree.getroot()
+        if tailoring_file_path is not None:
```

Comment:
You can reduce to a Pythonic `if not tailoring_file_path`, which would handle empty strings as well, and you could therefore streamline some of the higher-level code.